packages/cli/src/tools/loadMetroConfig.ts

```diff
@@ -15,11 +15,6 @@ function resolveSymlinksForRoots(roots: string[]): string[] {
   );
 }
 
-function getWatchFolders(): string[] {
-  const root = process.env.REACT_NATIVE_APP_ROOT;
-  return root ? resolveSymlinksForRoots([path.resolve(root)]) : [];
-}
```

Comment:
@thymikee also note that back in the day, we used to call `resovleSymlinksForRoots` in all cases - https://github.com/facebook/react-native/blob/0.55-stable/local-cli/util/Config.js#L54

Looks like this has changed and I don't know when. But it sounds like a regression.

Reply:
Okay, this has been done in this commit https://github.com/facebook/react-native/commit/893c0274d6c071c5fa99dee097e189441af74148 as a part of 0.57 release.

I am really confused right now because this change doesn't make any sense to me, unless Metro performs this functionally itself.

CC: @rafeca, I am hoping that you will be able to shed some light into this.

Reply:
Following up on this even further, this functionality has been introduced by @skevy in 2016 (lol) via https://github.com/facebook/react-native/commit/39f83c4a12b8350e331c3a6c575a2cab64950d7a#diff-d939fe299a50129184bf6b4208f6d497

Apologies for pinging both of you on such historical piece of code, but... does this mean symlinks used to work with React Native and could've been accidentally removed via https://github.com/facebook/react-native/commit/893c0274d6c071c5fa99dee097e189441af74148 commit?

I always believed that symlinks never worked and seeing symlinks related code always made me confused.

Reply:
Hey @grabbou ! 👋

It's hard to remember the intention of my commit there LOL (don't know if there's more context in D9444982 - I can't access diffs anymore), but it's quite probably that https://github.com/facebook/react-native/commit/893c0274d6c071c5fa99dee097e189441af74148 removed the `resolveSymlinksForRoots` call by mistake 😅

Reply:
CC: @rickhanlonii would you mind helping out and checking some context on the D9444982? Thanks @rafeca for answering!

Reply:
The only thing in the diff is this comment you left @rafeca:

> From what I see, we could actually just not set watchFolders, since right now it's only set to the value of REACT_NATIVE_APP_ROOT if it exists (which is an env variable that I cannot find in any RN docs).

I think we could safely revert the change as long as we confirm that https://github.com/facebook/react-native/issues/20712 doesn't regress 👍